### PR TITLE
[auth] Implement Reset Password screen (screens/auth/ForgotPasswordScreen)

### PR DESCRIPTION
**Pull request by allisonhongberkeley**
[//]: # "These comments are meant for your reference. They are invisible and don't need to be deleted!"

### Description
[//]: # "Required - Describe what's new in this PR in a few lines. A description and bullet points for specifics will suffice."

In this PR, I completed the frontend for the "Reset Password" screen and also implemented the password-reset functionality where clicking "send email" will generate a link to reset the password and email it to the user.

In addition, if the user inputs an incorrect email, an error message will appear. If the email validation is sent successfully, a success message will appear.

### Screenshots
[//]: # "Required for frontend changes, otherwise optional but strongly recommended. Add screenshots of expected behavior - GIFs if you're feeling fancy!"
IMG_9903
IMG_9904
IMG_9905

## How to review
[//]: # "Required - Describe the order in which to review files and what to expect when testing locally. Is there anything specifically you want feedback on? Should this be reviewed commit by commit, or all at once? What are some user flows to test? What are some edge cases to look out for?"

Go to the login page, click Forgot Password, and try sending yourself a reset email.

## Next steps
[//]: # "Optional - What's NOT in this PR, doesn't work yet, and/or still needs to be done. Note any temporary fixes in this PR that should be cleaned up later."

### Online sources
[//]: # "Optional - copy links to any tutorials or documentation that was useful to you when working on this PR"

https://firebase.google.com/docs/auth/web/manage-users#send_a_password_reset_email

### Related PRs
[//]: # "Optional - related PRs you're waiting on/ PRs that will conflict, etc; if this is a refactor, feel free to add PRs that previously modified this code"

[//]: # "This tags the project leader as a default. Feel free to change, or add on anyone who you should be in on the conversation."
🥦 CC: @sauhardjain

### CODE REVIEW

File: assets/Components.tsx [hunk 1]
```diff
@@ -74,3 +74,12 @@ export const RootNavBackButton = styled.TouchableOpacity`
...
 `;
+
+// export function NavigationHeaderContainer({ backgroundColor, alignItems });
+
+// NavigationHeaderContainer.propTypes = {
+
+//   backgroundColor: PropTypes.string,
+//   alignItems: PropTypes.string,
+
+// };
```
Comment (sauhardjain) [inline assets/Components.tsx]: We should remove this from the current PR.

File: src/screens/auth/ForgotPasswordScreen.tsx [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+  const handleSendEmail = async () => {
+    await forgotPassword(dispatch, { email });
+    setShowErrorMessage(true);
+    setShowSuccessMessage(true);
+  };
+
+  return (
+    <LoginContainer>
+      {/* logo placeholder */}
```
Comment (sauhardjain) [inline src/screens/auth/ForgotPasswordScreen.tsx]: Remove this comment.

File: src/screens/auth/styles.ts [hunk 3]
```diff
@@ -16,18 +16,15 @@ export const StartContainer = styled.View`
...
 `;
 
-export const LogoContainer = styled.View`
-  width: 100%;
-  justify-content: flex-start;
-  margin-left: 10%;
-  margin-top: 4%;
+export const ResetPasswordContainer = styled.View`
+  margin-top: 14%;
+  align-items: left;
```
Comment (sauhardjain) [inline src/screens/auth/styles.ts]: Where are you using this container?

File: src/screens/auth/styles.ts [hunk 4]
```diff
@@ -94,6 +91,18 @@ export const DarkGrayText = styled.Text`
   color: ${Colors.darkGray};
 `;
 
+export const BottomMargin = styled.View`
+  margin-bottom: 20px;
+`;
```
Comment (sauhardjain) [inline src/screens/auth/styles.ts]: This is not good practice for styling! Instead of having a bottom margin component, add the bottom margin/padding to the container.

File: src/screens/auth/styles.ts [hunk 5]
```diff
@@ -94,6 +91,18 @@ export const DarkGrayText = styled.Text`
...
+
+export const Header = styled.View`
+  width: 85%;
+  flex-direction: row;
+  justify-content: space-between;
+  align-items: center;
+  margin-bottom: 26px;
+`;
```
Comment (sauhardjain) [inline src/screens/auth/styles.ts]: Are you using the Header component in this PR?

File: src/utils/authUtils.ts [hunk 6]
```diff
@@ -16,6 +17,29 @@ export const setAuthErrorMessage = (
...
+) => {
+  const auth = getAuth();
+  await sendPasswordResetEmail(auth, params.email)
+    .then(() => {
+      setAuthSuccessMessage(dispatch, 'Email sent. Check your inbox to reset.');
+      setAuthErrorMessage(dispatch, '');
+    })
+    .catch(error => {
+      // eslint-disable-next-line no-console
+      console.warn('(forgotPassword) error', error);
```
Comment (sauhardjain) [inline src/utils/authUtils.ts]: Remove this console.warn since we're already catching the error using setAuthErrorMessage.

**Review by sauhardjain — CHANGES_REQUESTED** [review]
GREAT job Allison! Everything works well and the styling looks good too! 

Just a few nits on styling practices and some code from your sprint on creating the Header component that shouldn't be in this PR for cleanliness. Good to merge after you make these changes.

**Review by sauhardjain — APPROVED** [review]
